### ISSUE TITLE
Decoder with dummy classifier estimator always predict same label

### ISSUE THREAD

**Issue by NicolasGensollen** [nilearn/nilearn#2768]
It looks like the decoder with dummy classifier is always predicting the first class no matter the frequencies of the labels.

Here is an example using the Haxby dataset, where I change the target frequencies:

```python
import numpy as np
import pandas as pd
from numpy.random import choice
from collections import Counter
from nilearn.image import index_img
from nilearn.datasets import fetch_haxby
from nilearn.decoding import Decoder

# Load and pre-process data
haxby_dataset = fetch_haxby()
behavioral = pd.read_csv(haxby_dataset.session_target[0], delimiter=' ')
conditions = behavioral['labels']
# We consider 4 classes here
condition_mask = conditions.isin(['face', 'cat', 'bottle', 'shoe'])
data_subset = index_img(haxby_dataset.func[0], 
                        condition_mask)
conditions = conditions[condition_mask].values
print("Classes are initially perfectly balanced:")
print(Counter(conditions))
# Split data in train test for simplicity
train = index_img(data_subset, slice(0, -20)) 
test  = index_img(data_subset, slice(-20, None))
conditions_train = conditions[:-20]
conditions_test = conditions[-20:]
print("Dummy decoder predictions:")
dummy_decoder = Decoder(estimator='dummy_classifier', 
                        mask=haxby_dataset.mask_vt[0])
dummy_decoder.fit(train, 
                  conditions_train)
prediction = dummy_decoder.predict(test)
dummy_decoder.classes_[prediction]
```
```
Classes are initially perfectly balanced:
Counter({'face': 108, 'cat': 108, 'shoe': 108, 'bottle': 108})
Dummy decoder predictions:
array(['bottle', 'bottle', 'bottle', 'bottle', 'bottle', 'bottle',
       'bottle', 'bottle', 'bottle', 'bottle', 'bottle', 'bottle',
       'bottle', 'bottle', 'bottle', 'bottle', 'bottle', 'bottle',
       'bottle', 'bottle'], dtype='<U6')
```

```python
# Resample the labels to make bottle very unfrequent
# and face very frequent, such that we would expect the
# dummy decoder to always predict face rather than bottle
fake_conditions = choice(['face', 'cat', 'bottle', 'shoe'], 
                         len(conditions), 
                         p=[0.6, 0.2, 0.05, 0.15])
print("Resample the labels to make bottle very unlikely:")
print(Counter(fake_conditions))
fake_conditions_train = fake_conditions[:-20]
fake_conditions_test = fake_conditions[-20:]
print("Dummy decoder predictions:")
dummy_decoder = Decoder(estimator='dummy_classifier', 
                        mask=haxby_dataset.mask_vt[0])
dummy_decoder.fit(train, 
                  fake_conditions_train)
prediction = dummy_decoder.predict(test)
dummy_decoder.classes_[prediction]
```
```
Resample the labels to make bottle very unlikely:
Counter({'face': 265, 'cat': 93, 'shoe': 55, 'bottle': 19})
Dummy decoder predictions:
array(['bottle', 'bottle', 'bottle', 'bottle', 'bottle', 'bottle',
       'bottle', 'bottle', 'bottle', 'bottle', 'bottle', 'bottle',
       'bottle', 'bottle', 'bottle', 'bottle', 'bottle', 'bottle',
       'bottle', 'bottle'], dtype='<U6')
```

Maybe I'm missing something here, but I would expect a behavior like the following:

```python
from nilearn.input_data import NiftiMasker
from sklearn.dummy import DummyClassifier
d = DummyClassifier()
masker = NiftiMasker(haxby_dataset.mask_vt[0])
X_train = masker.fit_transform(train)
d.fit(X_train, 
      fake_conditions_train)
d.predict(masker.fit_transform(test))
```
```
array(['face', 'face', 'face', 'face', 'face', 'face', 'face', 'face',
       'face', 'face', 'face', 'face', 'face', 'face', 'face', 'face',
       'face', 'face', 'face', 'face'], dtype='<U4')
```

**Comment by NicolasGensollen** [comment]
I'm not quite sure yet, but I think this comes from the fact that we are binarizing the labels here:

https://github.com/nilearn/nilearn/blob/a36c0442fc8eebe401b310dd9799c0f071080e90/nilearn/decoding/decoder.py#L779-L782

For each binary classification problem where one class has label `1` and all others label `-1`, the dummy estimator always picks the same label `-1` since it is more frequent, unless we have very unbalanced classes such that label `1` is more frequent than `-1`.
